src/ingest/SpecRepo.cpp

```diff
@@ -151,15 +173,23 @@ void genSpecs4Roll(const std::string& version,
     // A roll spec will cover X days given table location of source data
     const auto now = Evidence::now();
     const auto max_days = table->max_hr / 24;
-    for (size_t i = 0; i <= max_days; ++i) {
-      // we only provide single macro for now
+    size_t i;
+    for (i = 0; i <= max_days; ++i) {
+      // run hourly partition list
+      for(size_t j = 0 ; j < DAY_HOURS; j++) {
+        genSpecs4HourlyRoll(version, table, 24, i, specs);
+      }
+      // TODO: refactor to have better support export from HMS chenqin
```

Comment:
sounds great - wondering what's the best way to plug it in. If you target this, could you please file an issue for it?

Reply:
genSpecs4HourlyRoll looks hacky, I updated pr with date/hour/minute/second granularity bucketing.
Ultimately, I am targeting to 
- scan and ingest smaller granularity bucket data into nebula (e.g minute level) periodically. 
- (pending discussion) offload one granularity higher parquet /csv/ orc file to upper granularity bucket (e.g hourly level) and put a _SUCCESS file when it complete.

Reply:
I'm looking at the new changes now - I'm assuming you meant some external process to do the work for your point number 2 listed above? Or you meant some internal auto roll-up process to transform lower granularity data to higher granularity? 

(internal roll-up could be a very interesting feature to have in Nebula but it's a big project deserved a nice design)

Reply:
We can start leverage external system and schedule corn jobs for now.
nebula customized roll-up indeed interesting from realtime serving->batch ingestion prospective. Most naive use case can be we have 5 mins partitioned data partitions ingested to nebula, nebula upload compacted parquet files to a hourly directory once observed timeValue greater than give hour.